internal/alerting/alerter.go

```diff
@@ -22,84 +18,41 @@ func (mn *MockAlerter) AlertError(e error) error {
 	return nil
 }
 
-func NewAlerter(cfg service.ErrorAlerting) (Alerter, error) {
+func NewAlerters(cfg service.ErrorAlerting) (Alerters, error) {
+	var alerters []Alerter
 	switch {
+	case cfg.Slack != nil:
+		alerter, err := NewSlackAlerter(cfg.Slack)
+		if err != nil {
+			return nil, err
+		}
+		alerters = append(alerters, alerter)
+		fallthrough
 	case cfg.PagerDuty != nil:
-		return NewPagerDutyAlerter(cfg.PagerDuty)
+		alerter, err := NewPagerDutyAlerter(cfg.PagerDuty)
+		if err != nil {
+			return nil, err
+		}
+		alerters = append(alerters, alerter)
 	}
-	return &MockAlerter{}, nil
-}
-
-type PagerDuty struct {
-	client     *pagerduty.Client
-	routingKey string
-}
 
-func NewPagerDutyAlerter(cfg *service.PagerDutyAlerting) (*PagerDuty, error) {
-	notifier := &PagerDuty{
-		client:     pagerduty.NewClient(cfg.ApiKey),
-		routingKey: cfg.RoutingKey,
-	}
-	if err := notifier.ping(); err != nil {
-		return nil, err
+	if len(alerters) == 0 {
+		return []Alerter{&MockAlerter{}}, nil
 	}
-	return notifier, nil
+
+	return alerters, nil
 }
 
-func (pd *PagerDuty) AlertError(e error) error {
+func (s *Alerters) AlertError(e error) error {
```

Comment:
We don't need the pointer receiver here. The type is defined as `[]Alerter` so there's no need for an pointer. It also lets us remove the dereference in the `for` loop below.